app/lib/rocketchat.js

```diff
@@ -776,15 +776,17 @@ const RocketChat = {
 	},
 	async getLoginServices(server) {
 		try {
-			let loginServicesFilter = [];
+			let loginServices = [];
 			const loginServicesResult = await fetch(`${ server }/api/v1/settings.oauth`).then(response => response.json());
 
 			if (loginServicesResult.success && loginServicesResult.services.length > 0) {
 				const { services } = loginServicesResult;
-				loginServicesFilter = services.filter(item => item.custom !== undefined); // TODO: remove this after SAML and CAS
+				loginServices = services;
+
+				const loginServicesReducer = loginServices.reduce((ret, item) => {
```

Comment:
You can use only `(services || []).reduce` and not need to create `loginServices`.

Reply:
True! We had a filter here, but we don't anymore. I'll fix it. Thanks :)

Reply:
After looking further into this, it's better this way.
`services` is a response from `settings.oauth` and we need to return its length.
This way we have one `return`... the other we would need two.
There's no right answer and both works.
I'll leave it as it is :)